drivers/staging/media/rpivid/rpivid_video.c

```diff
@@ -109,18 +113,31 @@ int rpivid_prepare_src_format(struct v4l2_pix_format_mplane *pix_fmt)
 	return 0;
 }
 
-int rpivid_prepare_dst_format(struct v4l2_pix_format_mplane *pix_fmt)
+/* Take any pix_format and make it valid */
+static void rpivid_prepare_dst_format(struct v4l2_pix_format_mplane *pix_fmt)
 {
 	unsigned int width = pix_fmt->width;
 	unsigned int height = pix_fmt->height;
 	unsigned int sizeimage = pix_fmt->plane_fmt[0].sizeimage;
 	unsigned int bytesperline = pix_fmt->plane_fmt[0].bytesperline;
 
-	switch (pix_fmt->pixelformat) {
+	if (!width)
+		width = RPIVID_DEFAULT_WIDTH;
+	if (width > RPIVID_MAX_WIDTH)
+		width = RPIVID_MAX_WIDTH;
+	if (!height)
+		height = RPIVID_DEFAULT_HEIGHT;
+	if (height > RPIVID_MAX_HEIGHT)
+		height = RPIVID_MAX_HEIGHT;
+
 	/* For column formats set bytesperline to column height (stride2) */
+	switch (pix_fmt->pixelformat) {
+	default:
+		pix_fmt->pixelformat = RPIVID_DEFAULT_DST_PIXELFORMAT;
```

Comment:
I'd actually make this explicitly V4L2_PIX_FMT_NV12_COL128, otherwise if someone changes RPIVID_DEFAULT_DST_PIXELFORMAT you still fall through into the NV12_COL128 case when it isn't.

Reply:
You have a point.  But I really want everything that is doing defaulting to change to the same thing - would it be OK to have the default case set pix_fmt & then goto the switch? It is mucky but it is the cleanest way of handling a changed default I can think of. (or check with #if and #error if not the expected value?)

Reply:
I'd be totally happy to validate and default if necessary the pixel format first, and then you handle alignments/widths for the supported formats.

Reply:
After going round a few circles trying to come up with something I believed in I've come back to your original suggestion of just setting it to V4L2_PIX_FMT_NV12_COL128.  I've also removed the definition of RPIVID_DEFAULT_DST_PIXELFORMAT as that was the only place it was used and having it would just add confusion.